### ISSUE TITLE
error in Preprocessing.encode_conditional_var(adata, column_id='sample')

### ISSUE THREAD

**Issue by Lucyyang1991** [SchubertLab/mvTCR#23]
I'm new to python, when I apply mvTCR to my own data, in the **Adding conditional variables** part, I ran `Preprocessing.encode_conditional_var(adata, column_id='sample')
`, and get an error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[17], [line 1](vscode-notebook-cell:?execution_count=17&line=1)
----> [1](vscode-notebook-cell:?execution_count=17&line=1) Preprocessing.encode_conditional_var(adata, column_id='sample')

File c:\users\a\mvtcr\mvtcr\utils_preprocessing.py:115, in check_if_input_is_mudata.<locals>.wrapper(mudata_gex_key, mudata_airr_key, *args, **kwargs)
    [113](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:113) 	train, test = func(*args, **kwargs)
    [114](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:114) else:
--> [115](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:115) 	func(*args, **kwargs)
    [116](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:116) #====================
    [117](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:117) #updating mudata
    [118](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:118) if input_is_mu and func_name in ("encode_clonotypes", "encode_tcr", "encode_conditional_var"):

File c:\users\a\mvtcr\mvtcr\utils_preprocessing.py:295, in Preprocessing.encode_conditional_var(adata, column_id)
    [287](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:287) @staticmethod
    [288](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:288) @check_if_input_is_mudata
    [289](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:289) def encode_conditional_var(adata, column_id):
    [290](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:290) 	"""
    [291](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:291) 	One-hot encode additional features in adata.obs. Column data will be transformed and the categories saved in adata.uns
    [292](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:292) 	:param adata: adata object
    [293](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:293) 	:param column_id: identifier for the specific column in the adata.obs for ohe
    [294](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:294) 	"""
--> [295](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:295) 	enc = OneHotEncoder(sparse=False)
    [296](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:296) 	enc.fit(adata.obs[column_id].to_numpy().reshape(-1, 1))
    [297](file:///C:/users/a/mvtcr/mvtcr/utils_preprocessing.py:297) 	adata.obsm[column_id + "_ohe"] = enc.transform(adata.obs[column_id].to_numpy().reshape(-1, 1))

TypeError: OneHotEncoder.__init__() got an unexpected keyword argument 'sparse'
```

Can you help me with this error?

**Comment by leuschjanphilipp** [comment]
Hello and thank you for trying out mvTCR!

Thanks for pointing that out, its a known bug and will be fixed in the next release. The reason you encounter this error is due to the sklearn package version in your environment. In version 1.2 sklearn changed the name of the argument for the one-hot encoder from `'sparse'` to `'sparse_output'`. 
A fix is to downgrade sklearn to v1.1.x with `pip install scikit-learn==1.1.3`. 
If you still encounter this or other issues or need further assistance, please let us know.

Cheers

**Comment by Lucyyang1991** [comment]
I solve this problem by change the utils_preprocessing.py file.
```
def encode_conditional_var(adata, column_id):
		"""
		One-hot encode additional features in adata.obs. Column data will be transformed and the categories saved in adata.uns
		:param adata: adata object
		:param column_id: identifier for the specific column in the adata.obs for ohe
		"""
		enc = OneHotEncoder(sparse_output=False)
		enc.fit(adata.obs[column_id].to_numpy().reshape(-1, 1))
		adata.obsm[column_id + "_ohe"] = enc.transform(adata.obs[column_id].to_numpy().reshape(-1, 1))
		adata.uns[column_id + "_enc"] = enc.categories_
```
